app/javascript/runtimeBanner/RuntimeBanner.jsx

```diff
@@ -0,0 +1,142 @@
+/* global Runtime */
+
+import { h } from 'preact';
+
+const BANNER_DISMISS_KEY = 'runtimeBannerDismissed';
+
+/**
+ * A banner that will be displayed to provide a deep link into a specified
+ * ConsumerApp based on the Runtime context. If the banner is dismissed it will
+ * keep track of this in localStorage so it won't be rendered again.
+ */
+export const RuntimeBanner = () => {
+  function dismissBanner() {
```

Comment:
Refactor (blocking): For `dismissBanner`, `removeFromDOM`, and `handleDeepLinkFallback` functions, define them outside the Preact function component as they do not require any variables from within the `RuntimeBanner` component.